sdk/core/azure-core/src/main/java/com/azure/core/util/QueryParameter.java

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Represents a single parameter to be added to a query string.
+ *<p>
+ * If multiple values are added to a query string with the same name (case-insensitive), then the values will be
+ * appended at the end of the same {@link QueryParameter} with commas separating them.
+ */
+public class QueryParameter {
+    private final String name;
+
+    // this is the actual internal representation of all values
+    private final List<String> values;
+
+    // but we also cache it to faster serve our public API
+    private String cachedStringValue;
+
+    /**
+     * Create a QueryParameter instance using the provided name and value.
+     *
+     * @param name the name of the parameter.
+     * @param value the value of the parameter.
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public QueryParameter(String name, String value) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.values = new LinkedList<>();
+        this.values.add(value);
+    }
+
+    /**
+     * Create a QueryParameter instance using the provided name and values.
+     *
+     * @param name the name of the parameter.
+     * @param values the values of the parameter.
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public QueryParameter(String name, String... values) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.values = new LinkedList<>();
+        for (String value : values) {
+            this.values.add(value);
+        }
+    }
+
+    /**
+     * Create a QueryParameter instance using the provided name and values.
+     *
+     * @param name the name of the parameter.
+     * @param values the values of the parameter.
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public QueryParameter(String name, List<String> values) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
```

Comment:
Should values be checked for being null? At least have a guard in the constructor as `LinkedList` will throw a NullPointerException if `values` is null

Reply:
Good catch. Solved.